CMakeLists.txt

```diff
@@ -95,11 +95,17 @@ option( E57_MAX_VERBOSE  "Compile library with maximum verbose logging" OFF )
 
 option( E57_WRITE_CRAZY_PACKET_MODE "Compile library to enable reader-stressing packets" OFF )
 
+option( E57_VISIBILITY_HIDDEN       "Compile library with hidden symbol visibility" OFF)
+
 #########################################################################################
 
 set( revision_id "${PROJECT_NAME}-${PROJECT_VERSION}-${${PROJECT_NAME}_BUILD_TAG}" )
 message( STATUS "[E57] Revision ID: ${revision_id}" )
 
+if ( E57_VISIBILITY_HIDDEN )
+    set(CMAKE_CXX_VISIBILITY_PRESET hidden)
```

Comment:
I think you should use the [target property](https://cmake.org/cmake/help/latest/prop_tgt/LANG_VISIBILITY_PRESET.html) `CXX_VISIBILITY_PRESET` so we don't pollute any including cmake contexts.

From the [docs](https://cmake.org/cmake/help/latest/module/GenerateExportHeader.html):

> The target properties CXX_VISIBILITY_PRESET and VISIBILITY_INLINES_HIDDEN can be used to add the appropriate compile flags for targets.

Reply:
Agreed, done.